paper/src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/basic/ArmourBound.java

```diff
@@ -72,6 +75,43 @@ public void onArmourEquip(PlayerArmorChangeEvent event){
 		}
 	}
 
+	@EventHandler
+	public void onItemCraft(PrepareItemCraftEvent event) {
+		CraftingInventory inventory = event.getInventory();
+		ItemStack[] matrix = inventory.getMatrix();
+		if (event.getRecipe() == null) {
+			return;
+		}
+		ItemStack boundItem = null;
+		PersistentDataContainer container = null;
+		ItemMeta meta = null;
+		for (ItemStack item : matrix) {
+			if (item == null) {
+				continue;
+			}
+			if (!whitelist.contains(item.getType().toString())) {
+				continue;
+			}
+			if (!item.hasItemMeta()) {
```

Comment:
Has functions are baaaaaad and encourage an extremely thread-unsafe style of coding. Deno for example has deprecated its `exists` function in its standard library because ["Checking the state of a file before using it causes a race condition. Perform the actual operation directly instead."](https://doc.deno.land/https://deno.land/std@0.145.0/fs/mod.ts/~/exists) It's best to code defensively, even if only for habitual reasons, which in this case would mean simply getting the item meta then testing whether it's null, that way you know *for certain* that the meta wouldn't have changed between the check and the retrieval.